src/components/quiz/QuizConstructor.vue

```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
+import EditButton from "../common/EditButton.vue";
+import _uniq from "lodash/uniq";
+import { computed, onMounted, ref } from "vue";
+import { QuizQuestion } from "../../../dto/quiz";
+import QuizList from "./QuizList.vue";
+
+defineProps({
+  question: {
+    type: Object as () => QuizQuestion,
+    default: () => ({}),
+  },
+});
+
+const selectedCategory = ref();
+const checkedQuestion = ref([]);
+const checkedAnswer = ref(0);
+
+onMounted(() => {
+  selectedCategory.value = "Select category for displaying questions";
+});
+
+const categories = computed(() =>
+  _uniq(questionList.map(item => item.category)),
+);
+
+const categoryQuestions = computed(() =>
+  questionList.filter(question => question.category === selectedCategory.value),
+);
+
+const quizList = ref<QuizQuestion[]>([]);
+
+const questionList = [
+  {
+    point: 1,
+    text: "text 1",
+    answer: "answer 1",
+    category: "HTML",
+    id: "1",
+  },
+  {
+    point: 5,
+    text: "text 2",
+    answer: "answer 2",
+    category: "HTML",
+    id: "2",
+  },
+  {
+    point: 5,
+    text: "text 3",
+    answer: "answer 3",
+    category: "CSS",
+    id: "3",
+  },
+  {
+    point: 5,
+    text: "text 4",
+    answer: "answer 4",
+    category: "CSS",
+    id: "4",
+  },
+  {
+    point: 1,
+    text: "text 6",
+    answer: "answer 6",
+    category: "Vue",
+    id: "6",
+  },
+  {
+    point: 1,
+    text: "text 7",
+    answer: "answer 7",
+    category: "JavaScript",
+    id: "7",
+  },
+];
+
+function addQuestions() {
+  const checkedArray = checkedQuestion.value;
```

Comment:
declaring of `checkedArray` is not needed you can do it like `quizList.value = _uniq([...quizList.value, ...checkedQuestion.value]);` or `quizList.value = quizList.value.concat(checkedQuestion.value)`